rustls-post-quantum/src/mlkem.rs

```diff
@@ -0,0 +1,96 @@
+use aws_lc_rs::kem;
+use aws_lc_rs::unstable::kem::ML_KEM_768;
+use rustls::crypto::{ActiveKeyExchange, CompletedKeyExchange, SharedSecret, SupportedKxGroup};
+use rustls::ffdhe_groups::FfdheGroup;
+use rustls::{Error, NamedGroup, PeerMisbehaved, ProtocolVersion};
+
+/// This is the [MLKEM] key exchange.
+///
+/// [MLKEM]: https://datatracker.ietf.org/doc/draft-connolly-tls-mlkem-key-agreement
+#[allow(clippy::upper_case_acronyms)]
+#[derive(Debug)]
+pub struct MLKEM768;
+
+impl SupportedKxGroup for MLKEM768 {
+    fn start(&self) -> Result<Box<dyn ActiveKeyExchange>, Error> {
+        let decaps_key = kem::DecapsulationKey::generate(&ML_KEM_768)
+            .map_err(|_| Error::FailedToGetRandomBytes)?;
+
+        let encaps_key = decaps_key
+            .encapsulation_key()
+            .map_err(|_| Error::FailedToGetRandomBytes)?;
+
+        let pub_key_bytes = encaps_key.key_bytes().unwrap();
+
+        Ok(Box::new(Active {
+            decaps_key: Box::new(decaps_key),
+            encaps_key_bytes: Vec::from(pub_key_bytes.as_ref()),
+        }))
+    }
+
+    fn start_and_complete(&self, client_share: &[u8]) -> Result<CompletedKeyExchange, Error> {
+        let (ciphertext, shared_secret) = kem::EncapsulationKey::new(&ML_KEM_768, client_share)
+            .map_err(|_| INVALID_KEY_SHARE)
```

Comment:
Suggest moving the `?` here and writing the `and_then()` as linear code instead of inside a combinator.